client/src/featureform/resources.py

```diff
@@ -1244,6 +1253,38 @@ def proto(self) -> pb.Columns:
 
 ResourceLocation = ResourceColumnMapping
 
+@typechecked
+@dataclass
+class TriggerResource:
```

Comment:
This should probably be a base class that can have different classes on top like ScheduleTrigger, WebhookTrigger, etc

Reply:
agreed